### DISCUSSION TITLE
How to create an Inverse Transform that is a mean of a distribution?

### DISCUSSION THREAD

**Discussion by ericjb** [alan-turing-institute/sktime#7331] [Q&A]
A point forecast is defined as the mean of a distribution. If my forecaster computes a forecast distribution for z = log(y), then the point forecast for y (i.e. in the original units) should be y_{t+h} = E[ exp( z_{t+h})]. In the case that the distribution of z_{t+h} is N(mu,sigma^2) this would lead to the point forecast y_{t+h} = exp(mu + sigma^2/2). In sktime, using the LogTransformer, the point forecast for y_{t+h} would be exp(mu) when using the "inverse transform" of the LogTransformer. If my forecaster (for log(y)) implements predict_var, then I would like to define the inverse transform to LogTransformer that takes the asymmetry of the back-transformation (exp) into account. How would one do this?

**Comment by fkiraly** [comment]
@ericjb, great you brought this up!

In my opinion, the cleanest way would be to introduce a transformed distribution object, and have the pipelines return that. The `predict_var` would then return the variance of the inverse transformed normal (in this case a log-normal).

I have also noticed the issue a while back and created some granular work items on `skpro` towards this, most are still open - the probability distribution issues do not get many contributors, as they require both math and software background.

Three related issues:

* https://github.com/sktime/skpro/issues/31 - this is almost identical to what you are requesting, but for tabular probabilistic regression and not probabilistic forecasting
* this composite distribution is required for it: https://github.com/sktime/skpro/issues/30
* related, more general: https://github.com/sktime/skpro/issues/321
* related, distributional forecast calibration: https://github.com/sktime/sktime/issues/6558

**Reply by fkiraly** [reply]
PS: feel free to open a feature request on `sktime` and link the issues. That would give the topic more visibility.

**Comment by ericjb** [comment]
Hi @fkiraly - this is the point I mentioned in today's discord meetup. There can be a debate whether this is a bug or, by adding a parameter for median or mean, can be viewed as a feature request. I know someone who may be interested in contributing to this project.
